### PR TITLE
[CLI-974] Tensorboard sync metrics cannot use relative time as the x-axis

### PR DESCRIPTION
**Pull request by peacekurella**
https://wandb.atlassian.net/browse/CLI-974

Description
-----------

This PR addresses a bug with the wandb client wherein `_runtime` was not always reported in RunHistory proto due to it being calculated in HistoryObjects, leading to inconsistencies when  syncing logs through tensorboard; this PR fixes the bug in the following ways:

1. Check to see if "_runtime" is already present in the run history when "handle_history" is called.
2. If it is not present, Inject "_runtime" by calling "_history_assign_runtime".
3. Proceed to calculate "_runtime" using "_timestamp", do nothing if "_timestamp" is not present.

Testing
-------

This fix was tested against two cases:

1. Using the script in function/sb3-integration with sync_tensorboard flag as True, ensuring that the live reporting doesn't break the runtime reporting
2.  Using the script in function/sb3-integration with sync_tensorboard flag as False, ensuring that the wandb sync doesn't break the runtime reporting

Release Notes
-------------

------------- BEGIN RELEASE NOTES ------------------

* Fixed a bug where the users can no longer use process time/wall time as the x-axis for metrics logged through tensorboard

------------- END RELEASE NOTES --------------------

### CODE REVIEW

**Review by vanpelt — COMMENTED** [review]
Cool, let's definitely write tests to assert the offline and online runtime calculations.

**Comment by peacekurella** [comment]
Links to the runs after the patch:
1. https://wandb.ai/peacekurella/CartPole-v1/runs/1hwgpx55 (live)
2. https://wandb.ai/peacekurella/CartPole-v1/runs/3dqhbj07 (wandb sync)

Link to a run before the patch:
1. https://wandb.ai/peacekurella/CartPole-v1/runs/329iccdj

**Comment by peacekurella** [comment]
@vanpelt to ensure test coverage for this, I am possibly thinking of adding a few assertions in "wandb_tensorflow_test.py" to ensure that:
1. All history dicts have a "_runtime" key
2. Ensure that the runtime is not in the same ballpark as the "_timestamp"

Let me know your thoughts on this.

**Review by vanpelt — APPROVED** [review]
Generally looks good.  Adding some assertions or new tests to the tensorboard side would be great.  We should check both offline sync and online mode.

**Review by dannygoldstein — COMMENTED** [review]
Looks good! Agree with @vanpelt  on testing. Also see my comment below about the current test. Why do we expect this to fail?

File: tests/test_offline_sync.py [hunk 1]
```diff
@@ -80,3 +80,15 @@ def test_sync_in_progress(live_mock_server, test_dir):
     assert len(glob.glob(os.path.join(latest_run, "*.synced"))) == 1
     print("Number of upserts: ", live_mock_server.get_ctx()["upsert_bucket_count"])
     assert live_mock_server.get_ctx()["upsert_bucket_count"] >= 3
+
+
+@pytest.mark.flaky
+@pytest.mark.xfail(reason="flaky test")
```
[the remark this answers is not in the document]
Comment (raubitsj) [reply tests/test_offline_sync.py:86]: flaky is just a mark, it doesnt do anything except annotate the test so you can find that this test is flaky
```
[pytest]
markers =
    image_output
    multiclass
    wandb_args
    flaky: tests marked as flaky need to be debugged
```

As for why this is flaky, i think we have many of our "wandb sync" tests being flaky when they are doing something more complex than this.

If this test is flaky, we should debug and fix it, there is nothing that should be non-deterministic about it.

**Review by raubitsj — APPROVED** [review]

**Pull request merged**